### PR TITLE
Add System.Security.Cryptography.Encoding Source and Tests

### PR DESCRIPTION
**Pull request by bartonjs**
Resolves https://github.com/dotnet/corefx/issues/833.

Will create an issue about OidInfo not matching Interop files style.

### CODE REVIEW

File: src/System.Security.Cryptography.Encoding/tests/OidCollection.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            i = c.Add(o0);
+            Assert.Equal(0, i);
+
+            Oid o1 = new Oid(SHA256_Oid, SHA256_Name);
+            i = c.Add(o1);
+            Assert.Equal(1, i);
+
+            Assert.Equal(2, c.Count);
+
+            Assert.True(Object.ReferenceEquals(o0, c[0]));
```
Comment (Maxwe11) [inline src/System.Security.Cryptography.Encoding/tests/OidCollection.cs]: `Assert.Same`

Comment (bartonjs) [inline src/System.Security.Cryptography.Encoding/tests/OidCollection.cs]: Good eye, fixed up the lot of them.

File: src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/AsnFormatter.Unix.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+{
+    internal abstract partial class AsnFormatter
+    {
+        private static readonly AsnFormatter s_instance = new OpenSslAsnFormatter();
+
+        // -----------------------------
+        // ---- PAL layer ends here ----
+        // -----------------------------
+    }
+}
```
Comment (stephentoub) [inline src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/AsnFormatter.Unix.cs]: Nit: some files are missing newlines at the end

**Comment by stephentoub** [comment]
LGTM

**Pull request merged**